backend/src/main/java/com/woowacourse/momo/member/service/MemberService.java

```diff
@@ -1,49 +1,71 @@
 package com.woowacourse.momo.member.service;
 
+import java.util.List;
+
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import lombok.RequiredArgsConstructor;
 
 import com.woowacourse.momo.auth.support.PasswordEncoder;
+import com.woowacourse.momo.global.exception.exception.ErrorCode;
+import com.woowacourse.momo.global.exception.exception.MomoException;
+import com.woowacourse.momo.group.domain.group.Group;
+import com.woowacourse.momo.group.service.GroupFindService;
 import com.woowacourse.momo.member.domain.Member;
-import com.woowacourse.momo.member.domain.MemberRepository;
 import com.woowacourse.momo.member.service.dto.request.ChangeNameRequest;
 import com.woowacourse.momo.member.service.dto.request.ChangePasswordRequest;
 import com.woowacourse.momo.member.service.dto.response.MemberResponseAssembler;
 import com.woowacourse.momo.member.service.dto.response.MyInfoResponse;
+import com.woowacourse.momo.participant.domain.ParticipantRepository;
 
 @RequiredArgsConstructor
 @Transactional(readOnly = true)
 @Service
 public class MemberService {
 
     private final MemberFindService memberFindService;
-    private final MemberRepository memberRepository;
     private final PasswordEncoder passwordEncoder;
+    private final GroupFindService groupFindService;
+    private final ParticipantRepository participantRepository;
 
     public MyInfoResponse findById(Long id) {
-        Member member = memberFindService.findMember(id);
+        Member member = memberFindService.findExistMember(id);
 
         return MemberResponseAssembler.myInfoResponse(member);
     }
 
     @Transactional
     public void deleteById(Long id) {
-        memberRepository.deleteById(id);
+        Member member = memberFindService.findExistMember(id);
+        leaveGroup(member);
+        member.delete();
+    }
+
+    private void leaveGroup(Member member) {
+        List<Group> groups = groupFindService.findRelatedGroups(member.getId());
+        for (Group group : groups) {
+            if (group.isEnd()) {
+                continue;
+            }
+            if (group.isHost(member)) {
+                throw new MomoException(ErrorCode.MEMBER_DELETED_EXIST_IN_PROGRESS_GROUP);
+            }
+            participantRepository.deleteByGroupIdAndMemberId(group.getId(), member.getId());
+        }
     }
```

Comment:
```suggestion
    @Transactional
    public void deleteById(Long id) {
        Member member = memberFindService.findExistMember(id);
        
        leaveProgressingGroups(member);
        
        member.delete();
    }

    private void leaveProgressingGroups(Member member) {
        List<Group> progressingGroups = groupFindService.findRelatedGroups(member.getId())
                .stream()
                .filter(group -> !group.isEnd())
                .collect(Collectors.toList());

        validateMemberNotHost(member, progressingGroups);

        progressingGroups.forEach(
                group -> participantRepository.deleteByGroupIdAndMemberId(group.getId(), member.getId()));
    }

    private void validateMemberNotHost(Member member, List<Group> groups) {
        if (isMemberHost(member, groups)) {
            throw new MomoException(ErrorCode.MEMBER_DELETED_EXIST_IN_PROGRESS_GROUP);
        }
    }

    private boolean isMemberHost(Member member, List<Group> groups) {
        return groups.stream()
                .anyMatch(group -> group.isHost(member));
    }
```

저라면 이렇게 리팩터링했을 듯 합니다